File: TugasModul4/src/ros_service_cpp/src/server_node.cpp
#include "ros/ros.h"
#include "ros_service_cpp/magnitude.h"

bool vector_magnitude(ros_service_cpp::magnitude::Request &req, ros_service_cpp::magnitude::Response &resp){
    resp.magnitude = sqrt(req.komponenX*req.komponenX + req.komponenY*req.komponenY + req.komponenZ*req.komponenZ);
    return true;
}

int main(int argc, char **argv){
    ros::init(argc, argv, "service_server");
    ros::NodeHandle nh;

    ros::ServiceServer server = nh.advertiseService("magnitude_cpp", vector_magnitude);
    ROS_INFO("siap perhitungan.");
    ros::spin();

    return 0;
}

File: TugasModul4/src/ros_service_python/script/server_node.py
#!/usr/bin/env python3
import rospy
import math as mt
from ros_service_python.srv import magnitude, magnitudeResponse

def vector_magnitude(req):
    return magnitudeResponse(mt.sqrt(req.komponenX**2 + req.komponenY**2 + req.komponenZ**2))

def server_node():
    rospy.init_node("server")

    s = rospy.Service('menghitung_magnitude_py', magnitude, vector_magnitude)
    rospy.loginfo("siap menghitung")
    rospy.spin()
    
if __name__ == '__main__':
    server_node()
